metaci/testresults/importer.py

```diff
@@ -56,20 +60,18 @@ def import_test_results(build_flow, results):
 
         testclass = classes.get(result['ClassName'], None)
         if not testclass:
-            try:
-                testclass = TestClass.objects.get(name = result['ClassName'], repo=build_flow.build.repo)
-            except TestClass.DoesNotExist:
-                testclass = TestClass(name = result['ClassName'], repo=build_flow.build.repo)
-                testclass.save()
+            testclass, created = TestClass.objects.get_or_create(
+                name = result['suite']['name'],
+                repo=build_flow.build.repo,
+                test_type='Apex',
+            )
             classes[result['ClassName']] = testclass
 
         method = methods.get(class_and_method, None)
         if not method:
-            try:
-                method = TestMethod.objects.get(testclass = testclass.id, name = result['Method'])
-            except TestMethod.DoesNotExist:
-                method = TestMethod(testclass = testclass, name = result['Method'])
-                method.save()
+            method, created = TestMethod.objects.get_or_create(
+                testclass = testclass, name = result['name']
```

Comment:
Ditto

Reply:
I'm not sure what's the issue here.  TestMethod doesn't have a test_type field on it.  The only thing I changed was to use get_or_create instead of try/except DoesNotExist.